.circleci/workflows.yml

```diff
@@ -763,62 +779,86 @@ jobs:
             - *authenticate
-            - add_ssh_keys:
-                # deploy key to private-bigquery-etl
-                fingerprints:
-                  - "9d:1e:af:52:78:2c:e8:ec:33:4c:db:cd:5a:ff:70:0a"
-            - run:
-                name: Install rsync
-                command: |
-                  apt update
-                  apt install -y rsync
+            - *add_private_bigquery_etl_ssh_keys
             - run:
                 name: Pull down private SQL content
                 command: |
                   ssh-keyscan -t rsa github.com >> ~/.ssh/known_hosts
-                  git clone --single-branch --branch main \
-                                git@github.com:mozilla/private-bigquery-etl.git \
-                                ~/private-bigquery-etl
-                  rsync --archive ~/private-bigquery-etl/sql/ sql/
-                  rsync --archive ~/private-bigquery-etl/dags.yaml dags.yaml
+                  git clone --branch main --depth 1 \
+                    git@github.com:mozilla/private-bigquery-etl.git ~/private-bigquery-etl
             - run:
                 name: Generate SQL content
-                no_output_timeout: 30m
                 command: |
-                  mkdir -p /tmp/workspace/private-generated-sql
+                  export PATH="$PWD/venv/bin:$PATH"
+                  export PYTHONPATH="$PWD:$PYTHONPATH"
+                  bqetl_script="$PWD/script/bqetl"
+                  cd ~/private-bigquery-etl
 
-                  cp -r sql/ /tmp/workspace/private-generated-sql/sql
-                  # Don't depend on dry run for PRs
-                  PATH="venv/bin:$PATH" script/bqetl generate all \
-                    --output-dir /tmp/workspace/private-generated-sql/sql/ \
-                    --target-project moz-fx-data-shared-prod
-                  PATH="venv/bin:$PATH" script/bqetl dependency record \
-                    --skip-existing \
-                    "/tmp/workspace/private-generated-sql/sql/"
-                  PATH="venv/bin:$PATH" script/bqetl metadata update \
-                    --sql-dir /tmp/workspace/private-generated-sql/sql/ \
-                    /tmp/workspace/private-generated-sql/sql/
-                  PATH="venv/bin:$PATH" script/bqetl format /tmp/workspace/private-generated-sql/sql/
-                  PATH="venv/bin:$PATH" script/bqetl monitoring update /tmp/workspace/private-generated-sql/sql/
+                  $bqetl_script generate all --target-project moz-fx-data-shared-prod
+                  $bqetl_script dependency record --skip-existing sql/
+                  $bqetl_script metadata update sql/
+                  $bqetl_script format sql/
+                  $bqetl_script monitoring update sql/
 
+                  mkdir -p /tmp/workspace/private-generated-sql
+                  cp -r /tmp/workspace/generated-sql/sql /tmp/workspace/private-generated-sql/sql
+                  cp -rf sql/. /tmp/workspace/private-generated-sql/sql
```

Comment:
should this go before `bqetl dependency record` so that it records the cross-repo depencies or does that not matter?

Reply:
That doesn't matter, as `bqetl dependency record` is only extracting direct table/view references from the SQL files, not recursively finding table references through referenced views.